fwoa-tools/src/binaryConverter.test.ts

```diff
@@ -0,0 +1,299 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as fs from 'fs';
+import AWS from 'aws-sdk';
+import { GetObjectRequest, ListObjectsV2Request, PutObjectRequest } from 'aws-sdk/clients/s3';
+import * as AWSMock from 'aws-sdk-mock';
+import {
+  convertBinaryResource,
+  getBinaryObject,
+  getBinaryResource,
+  parseCmdOptions,
+  uploadBinaryResource
+} from './binaryConverter';
+import { ExportOutput } from './migrationUtils';
+
+AWSMock.setSDKInstance(AWS);
+let fakeFile: ExportOutput = {
+  jobId: 'binaryUnitTests',
+  file_names: { Binary: ['Binary/Binary-0.ndjson'] }
+};
+
+describe('binaryConverter', () => {
+  afterEach(() => {
+    AWSMock.restore();
+  });
+
+  beforeEach(() => {
+    jest.resetModules();
+    process.env.BINARY_BUCKET_NAME = 'unit_test_binary_bucket_name';
+    process.env.EXPORT_BUCKET_NAME = 'unit_test_export_bucket_name';
+  });
+
+  describe('parseCmdOptions', () => {
+    beforeEach(() => {
+      process.env.BINARY_BUCKET_NAME = 'unit_test_binary_bucket_name';
+      process.env.EXPORT_BUCKET_NAME = 'unit_test_export_bucket_name';
+    });
+
+    test('dryrun enabled', () => {
+      process.argv = ['/usr/local/bin/ts-node', 'binaryConverter.ts', '-d'];
+
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      const argv: any = parseCmdOptions();
+      expect(argv.dryRun).toEqual(true);
+    });
+    test('dryrun not enabled', () => {
+      process.argv = ['/usr/local/bin/ts-node', 'binaryConverter.ts'];
+
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      const argv: any = parseCmdOptions();
+      expect(argv.dryRun).toEqual(false);
+    });
+  });
+
+  describe('convertBinaryResource', () => {
+    let logs: fs.WriteStream;
+    beforeAll(() => {
+      logs = fs.createWriteStream('binary_unit_test.log');
+    });
+
+    beforeEach(() => {
+      process.env.BINARY_BUCKET_NAME = 'unit_test_binary_bucket_name';
```

Comment:
We should restore the original environment env after each run, example: https://github.com/aws-solutions/fhir-works-on-aws/blob/develop/fwoa-tools/src/migrationUtils.test.ts#L22

Reply:
Added a line to restore env after all tests